src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract IImmutableList<SyntaxNode> GetAssemblyScopedAttributeSyntaxNodesOfDocument(SyntaxNode documentRoot);
+        protected abstract SyntaxNode GetConstructorArgumentOfInternalsVisibleToAttribute(SyntaxNode internalsVisibleToAttribute);
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '\"';
```

Comment:
Ah, you're right about `IsStartingNewWord`. My bad, should have read the whole method :).

My concern was for users who didn't trigger on the opening double quote, but started typing the 
string. So they start with 

`[InternalVisiblesTo("$$")`

and start typing characters at the $$. In that case, we wouldn't trigger. I guess to make this really simple, we can check to see if you're in one of these two situations:

- `[InternalVisiblesTo("$$")`
- `[InternalVisiblesTo("a$$")`

You've already implemented the first one, and the second one would be a matter of checking if the character at caretPosition is a valid assembly name start charcter and that the character at caretPosition-1 is a double quote. How does that sound? That lets us avoid avoid having to deal with triggering in the middle of the string, which you noted we already don't do.